src/Test/DocTest.hs

```diff
@@ -117,14 +122,13 @@ filterModules wantedMods0 allMods0
   nonExistingMods = Set.toList (wantedMods1 `Set.difference` allMods1)
   isSpecifiedMod Module{moduleName} = moduleName `Set.member` wantedMods1
 
-setSeed :: Bool -> ModuleConfig -> IO ModuleConfig
-setSeed quiet cfg@ModuleConfig{cfgRandomizeOrder=True, cfgSeed=Nothing} = do
+setSeed :: (?verbosity :: LogLevel) => ModuleConfig -> IO ModuleConfig
```

Comment:
I guess it wasn't worth making a dedicated monad?  (Momentarily, it would be implemented as `ReaderT LogLevel IO`.)
I used implicit args once to not have to convert pure code to monadic code, but since you are already dealing with monadic code, the case for `ImplicitParams` is weaker.  (I think it is part of the GHC feature bloat that was maybe easy to add but only sees fringe use.)
Anyway, I am just poking my curious nose in here, feel free to ignore.

Reply:
I don't really like the ergonomics of Monad stacks, which is why I do like `ImplicitParams`: named arguments and "free" stacking is really something Haskell misses out on.

>  Anyway, I am just poking my curious nose in here, feel free to ignore.

No worries :)